ngraph/core/src/op/tile.cpp

```diff
@@ -115,6 +115,12 @@ bool op::v0::Tile::evaluate(const HostTensorVector& outputs, const HostTensorVec
     {
         output_shape[i] = data_shape[i] * repeats_val[i];
     }
+
+    if (!output->get_is_allocated())
```

Comment:
Why do we need this code? I thought that HostTensor already should be allocated.

Reply:
This was actually a surprise for me when I started segfaulting in one of the Tile operator tests. The `HostTensor`s are lazily allocated https://github.com/openvinotoolkit/openvino/blob/master/ngraph/core/src/runtime/host_tensor.cpp#L130 and this started happening when I changed the CF pass. 
CF pass now does the revalidation of nodes and dynamic Tiles are changed to static so it should not happen any more but this code protects from segfaults if the Tile's evaluator is used from somewhere else than CF.

Reply:
Should we add new test to cover this code? Or is it already covered?

Reply:
Also am I right that we can have the same issue in any operation? Should we generalize it?

Reply:
I was wrong, it doesn't cause a segfault. In the worst case an exception is thrown:
```
[ RUN      ] INTERPRETER.onnx_dyn_shapes_model_tile
unknown file: Failure
C++ exception with description "Check 'get_partial_shape().is_static()' failed at ngraph/core/src/runtime/host_tensor.cpp:78:
Attempt to allocate buffer for tensor with partial shape: ?
" thrown in the test body.
[  FAILED  ] INTERPRETER.onnx_dyn_shapes_model_tile (11 ms)
```

In case of tile we're lucky because we can try to evaluate this op even if the node has dynamic output shape. We have enough information in the input tensors passed to the evaluate method to calculate it.

It can probably be done for some other ops but it requires some analysis and it's op-by-op analysis in this case.